<launch>
  <arg name="dxl_usb_port"            default="/dev/ttyUSB0"/>
  <arg name="dxl_baud_rate"           default="4000000"/>
  <arg name="loop_rate"               default="200"/>

  <arg name="only_monitor"            default="False"/>
  <arg name="fake_mode"               default="True"/>

  <arg name="initial_pose"            default="stand"/>
  <arg name="initial_fake_pose"       default="rest_air"/>
  <arg name="init_pose_step"          default="0.001"/>

  <param name="dynamixel_info"        value="$(find k3lso_dynamixel)/config/k3lso_config.yaml"/>
  <param name="k3lso_poses_yaml"      value="$(find k3lso_dynamixel)/config/k3lso_poses.yaml"/>

  <node name="k3lso_dynamixel" pkg="k3lso_dynamixel" type="k3lso_dynamixel.py" required="true" output="screen">
    <param name="dxl_usb_port"        value="$(arg dxl_usb_port)"/>
    <param name="dxl_baud_rate"       value="$(arg dxl_baud_rate)"/>
    <param name="loop_rate"           value="$(arg loop_rate)"/>
    <param name="only_monitor"        value="$(arg only_monitor)"/>
    <param name="fake_mode"           value="$(arg fake_mode)"/>
    <param name="initial_pose"        value="$(arg initial_pose)"/>
    <param name="initial_fake_pose"   value="$(arg initial_fake_pose)"/>
    <param name="init_pose_step"      value="$(arg init_pose_step)"/>
  </node>
</launch>